src/postformatting/rule-selector-remover.js

```diff
@@ -0,0 +1,33 @@
+import csstree from 'css-tree'
+
+export default function ruleSelectorRemover (ast, selectorNodeMap, selectors) {
+  selectors = new Set(selectors)
+
+  csstree.walk(ast, {
+    visit: 'Rule',
+    enter: function (rule, item, list) {
+      // remove a rule with bad selector
+      if (rule.prelude.type !== 'SelectorList') {
+        list.remove(item)
+        return
+      }
+
+      // check what, if any selectors are found above fold
+      // filter out the ones that are not critical
+      rule.prelude.children = rule.prelude.children.filter(
+        (selectorNode, item, list) => {
+          let decision = selectorNodeMap.get(selectorNode)
+          if (typeof decision === 'string') {
+            decision = selectors.has(decision)
+          }
+          return typeof decision !== 'boolean' || decision
+        }
+      )
+
+      // NOTE: isCssSelectorRuleCritical mutates the rule
```

Comment:
This comment is in the wrong place now, I think it can just be removed?

Reply:
Hm, looks like you comment an outdated source, because I remember that I deleted this comment

Reply:
Yep indeed, I looked at the commits one by one - and I already fixed them in master too, so please just ignore (the ones about changes). All good! 👍